pkg/metrics/opentelemetry.go

```diff
@@ -30,6 +30,74 @@ var (
 	openTelemetryGaugeCollector *gaugeCollector
 )
 
+// SetMeterProvider sets the meter provider for the tests.
+func SetMeterProvider(meterProvider *sdkmetric.MeterProvider) {
```

Comment:
I think this method can be removed, as the call to `otel.SetMeterProvider(meterProvider)` should be enough for testing.